modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/MinimumRequiredTimeCollectorService.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.table.distributed.raft;
+
+import java.util.Map;
+import org.apache.ignite.internal.replicator.TablePartitionId;
+
+/** Collects minimum required timestamp for each partition. */
+public interface MinimumRequiredTimeCollectorService {
+
+    /** Undefined value of a min timestamp. */
+    long UNDEFINED_MIN_TIME = 0;
+
+    /** Registers a partition. */
+    void addPartition(TablePartitionId tablePartitionId);
+
+    /** Records the given timestamp .*/
+    void recordMinActiveTxTimestamp(TablePartitionId tablePartitionId, long timestamp);
+
+    /** Remove timestamps associated with the given partition .*/
+    void removePartition(TablePartitionId tablePartitionId);
+
+    /** Returns a snapshot of collected timestamps. */
+    Map<TablePartitionId, Long> minTimestampPerPartition();
```

Comment:
This component only deals with `TablePartitionId` and callers of this method check their `TablePartitionId`s with results of this method.

Reply:
I mean that within collocation track zones instead of tables will manage paritions, thus ZonePartitionId will be used. So that if it's possible to use ReplicationGroupId in the component, it's better to do that.

Reply:
How can I obtain zoneId inside PartitionListener then?

Reply:
Discussed in person.